src/discordtipbot/discordbot.cr

```diff
@@ -517,6 +519,12 @@ class DiscordBot
     reply(msg, "Bot has been running for #{Time.now - START_TIME}")
   end
 
+  def stats(msg : Discord::Message)
+    guilds = @cache.guilds.size
+    users = @cache.users.size
```

Comment:
This only reflects the number of users you happen to have cached in memory (typically only online users). A more useful stat might be `@guilds.values.map(&.member_count).sum` and describing it as "approximate users". Sure this does not account for the same user in multiple guilds, but it is very rare that a user is in more than one of the guilds your bot is in in the bigger picture.

Reply:
Hm.
Does `@cache.guilds` always contain all guilds? Else I could maybe make it `~ X guilds (cached: Y) and ~ A users (cached: B)

Reply:
All guilds the shard is connected to is always received after `READY`, but `GUILD_CREATE` only contains online user presences at the time of startup.

Reply:
After 100 members I assume. Else the count would be currently off.

Reply:
```
  def stats(msg : Discord::Message)
    guilds = @cache.guilds.size
    cached_users = @cache.users.size
    users = @cache.guilds.values.map(&.member_count).sum

    reply(msg, "The bot is in #{guilds} Guilds and sees #{users} users (of which #{cached_users} users are guaranteed unique)")
  end```

Reply:
@z64 does this look good?

Reply:
Up to you how you want to handle it. `@cache.users.size` ultimately can't be relied on (the error will only grow with the number of guilds your on, with more error and less reliability than summing `member_count`)